featureflags/http/managers/httpx.py

```diff
@@ -0,0 +1,166 @@
+        "like this `pip install 'featureflags-client[httpx]'`"
+    ) from None
+
+log = logging.getLogger(__name__)
+
+
+class AsyncHttpManager(AsyncAbstractManager):
+    """Feature flags manager for asyncio apps with `httpx` client."""
+
+    def __init__(  # noqa: PLR0913
+        self,
+        url: str,
+        project: str,
+        variables: List[Variable],
+        defaults: Union[EnumMeta, Type, Mapping[str, bool]],
+        *,
+        request_timeout: int = 5,
+        refresh_interval: int = 10,
+    ) -> None:
+        super().__init__(
+            url,
+            project,
+            variables,
+            defaults,
+            request_timeout,
+            refresh_interval,
+        )
+
+        self._session = httpx.AsyncClient(base_url=url)
+        self._refresh_task: Optional[asyncio.Task] = None
+
+    async def preload(self) -> None:
+        """
+        Preload flags from the server.
+        """
+
+        payload = PreloadFlagsRequest(
+            project=self._state.project,
+            variables=self._state.variables,
+            flags=self._state.flags,
+            version=self._state.version,
+        )
+        log.debug(
+            "Exchange request, project: %s, version: %s, flags: %s",
+            payload.project,
+            payload.version,
+            payload.flags,
+        )
+
+        response_raw = await self._post(
+            url=httpx.URL(Endpoints.PRELOAD.value),
+            payload=asdict(payload, dict_factory=custom_asdict_factory),
+            timeout=self._request_timeout,
+        )
+        log.debug("Preload response: %s", response_raw)
+
+        response = PreloadFlagsResponse.from_dict(response_raw)
+        self._state.update(response.flags, response.version)
+
+    async def sync(self) -> None:
+        payload = SyncFlagsRequest(
+            project=self._state.project,
+            flags=self._state.flags,
+            version=self._state.version,
+        )
+        log.debug(
+            "Sync request, project: %s, version: %s, flags: %s",
+            payload.project,
+            payload.version,
+            payload.flags,
+        )
+
+        response_raw = await self._post(
+            url=httpx.URL(Endpoints.SYNC.value),
+            payload=asdict(payload, dict_factory=custom_asdict_factory),
+            timeout=self._request_timeout,
+        )
+        log.debug("Sync reply: %s", response_raw)
+
+        response = SyncFlagsResponse.from_dict(response_raw)
+        self._state.update(response.flags, response.version)
+
+    def get(self, name: str) -> Callable[[Dict], bool] | None:
+        return self._state.get(name)
+
+    def start(self) -> None:
+        if self._refresh_task is not None:
+            raise RuntimeError("Manager is already started")
+
+        self._refresh_task = asyncio.create_task(self._refresh_loop())
+
+    async def wait_closed(self) -> None:
+        self._refresh_task.cancel()
```

Comment:
In grpc manager, there is a separate method `close` which closes refresh_task. Lets try following that implementation. (Or if all in one method is desired we can rename it to `async close`)